### PR TITLE
Initial e2e pack-peel pipeline through MLIR-AIR

### PR DESCRIPTION
**Pull request by erwei-xilinx**

### CODE REVIEW

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.h [hunk 1]
```diff
@@ -16,7 +16,7 @@ namespace mlir::iree_compiler::AMDAIE {
 
 /// Add passes to lower from MLIR-AIR through AIE. This is
 /// currently the default passes used for lowering after IREEs tiling.
-void addMLIRAIRAIELoweringPasses(OpPassManager &passManager);
+void addMLIRAIRAIELoweringPasses(OpPassManager &passManager, bool packPeel);
```
Comment (yzhang93) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.h:19]: Please add some comments about the new argument. Is this something temporary?

File: tests/samples/pack_peel_pipeline.mlir [hunk 2]
```diff
@@ -10,41 +10,34 @@ func.func @matmul_example(%lhs: tensor<1024x512xi8>, %rhs: tensor<512x1024xi8>)
...
+                    outs(%1: tensor<512x512xbf16>) -> tensor<512x512xbf16>
+  return %res : tensor<512x512xbf16>
+}
+
+// CHECK-LABEL: hal.executable.export public @matmul_large_dispatch_0_matmul_512x512x1024_bf16
+//       CHECK:    aie.device(ipu)
+//       CHECK:    aie.shim_dma_allocation
+//       CHECK:    aie.shim_dma_allocation
+//       CHECK:    aie.shim_dma_allocation
+//       CHECK:    func.func @matmul_large_dispatch_0_matmul_512x512x1024_bf16(%arg0: memref<262144xi32>, %arg1: memref<262144xi32>, %arg2: memref<131072xi32>)
```
Comment (yzhang93) [inline tests/samples/pack_peel_pipeline.mlir:39]: Why are the arguments i32 type?

File: tests/samples/pack_peel_pipeline.mlir [hunk 3]
```diff
@@ -10,41 +10,34 @@ func.func @matmul_example(%lhs: tensor<1024x512xi8>, %rhs: tensor<512x1024xi8>)
...
+//       CHECK:    aie.shim_dma_allocation
+//       CHECK:    aie.shim_dma_allocation
+//       CHECK:    aie.shim_dma_allocation
+//       CHECK:    func.func @matmul_example_dispatch_0_matmul_1024x1024x512_i8xi8xi32(%arg0: memref<131072xi32>, %arg1: memref<131072xi32>, %arg2: memref<1024x1024xi32>)
+//       CHECK:      aiex.ipu.dma_memcpy_nd
+//       CHECK:      aiex.ipu.dma_memcpy_nd
+//       CHECK:      aiex.ipu.dma_memcpy_nd
+//       CHECK:      aiex.ipu.sync
+
+func.func @matmul_large(%lhs: tensor<512x1024xbf16>, %rhs: tensor<1024x512xbf16>) -> tensor<512x512xbf16>
```
Comment (yzhang93) [inline tests/samples/pack_peel_pipeline.mlir:24]: This is not a large matmul size, maybe you want to test bf16?

**Review by yzhang93 — CHANGES_REQUESTED** [review]
Also we may want to add some tests to ci https://github.com/nod-ai/iree-amd-aie/blob/main/build_tools/ci/run_matmul_test.sh.

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.h (hunk 1 above)
Comment (erwei-xilinx) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.h:19]: Added comment to Passes.cpp. Yes eventually we would use the same pipeline as pad-pack. Please see comments inlined for the current pipeline differences.

File: tests/samples/pack_peel_pipeline.mlir (hunk 2 above)
Comment (erwei-xilinx) [reply tests/samples/pack_peel_pipeline.mlir:39]: This has always been like this in the workflow. IPU SHIM DMA BD's wrap and stride fields count in number of 32-bit words. So AIR is packing data with datatype narrower than 32 bits to 32-bit words, before lowering to IPU instructions.

File: tests/samples/pack_peel_pipeline.mlir (hunk 3 above)
Comment (erwei-xilinx) [reply tests/samples/pack_peel_pipeline.mlir]: Test renamed.

**Review by yzhang93 — APPROVED** [review]
Thanks!

**Pull request merged**